numba/cpython/builtins.py

```diff
@@ -595,3 +595,17 @@ def impl(func, iterable):
                 if func(x):
                     yield x
     return impl
+
+# ------------------------------------------------------------------------------
+# str overload
+
+@overload(str)
+def ol_str(s):
+    def impl(s):
+        return s.__str__()
+    return impl
```

Comment:
This would probably be better off as a lambda too.

Reply:
Just wanted to mention that in the long term (outside the scope of this PR), the global overload for the built-in `str` should be expanded to call `__repr__` if `__str__` is not available (haven't figured out how to do this yet since the `hasattr` and `getattr` built-ins aren't yet implemented.

Reply:
I think we should defer this part of the change set to https://github.com/numba/numba/pull/5866 and for this case just overload `str`, guard against any type other than `Integer`, and implement as already presented herein. Alternative is to just wait until that is done, but I'd rather get this in for 0.51.0.

@AndrewEckart RE how to do the thing you are asking, I'll comment on your PR.

Thanks all.

Reply:
Ok, I will undo those changes